### ISSUE TITLE
Mount not being removed after deletion

### ISSUE THREAD

**Issue by garyb** [slamdata/slamengine#737]
Here's my config:

``` json
{
  "server": {
    "port": 8080
  },
  "mountings": {
    "/data1": {
      "mongodb": {
        "connectionUri": "mongodb://localhost:27001"
      }
    },
    "/data2": {
      "mongodb": {
        "connectionUri": "mongodb://localhost:27002"
      }
    }
  }
}
```

- I `DELETE` `/mount/fs//data2/` and get a 200 response `deleted /data2/`
- The server restarts
- The `config.json` timestamp is updated

But the `config.json` contents is unchanged and both mounts are still present.

**Comment by sellout** [comment]
So yeah, the behavior here is bad.

Currently, we return 200 if we couldn’t find the resource (since, hey, the old state is what you wanted, right?) but we should probably return 404 in that case. Then at least you’d know that it didn’t find the thing you thought you were deleting.

**Comment by sellout** [comment]
Ok, the deletion behavior was broken. The 404 and 200 cases were reversed – and that’s why it’s especially confusing in #739.

I’m treating this ticket as “broken DELETE” (which I’ve already fixed), and the / no-/ behavior will be fixed with #739.
